jetty-security/src/main/java/org/eclipse/jetty/security/HashLoginService.java

```diff
@@ -126,14 +127,56 @@ public void setHotReload(boolean enable)
         this.hotReload = enable;
     }
 
+    /**
+     * Will we automatically start/stop the user store?
+     *
+     * @return true if doStart and doStop will call start/stop on the user store
+     */
+    public boolean isStartUserStoreEnabled()
+    {
+        return startUserStore;
+    }
+
+    /**
+     * Enable calling start/stop on the configured user store.
+     *
+     * @param enable true to enable, false to disable
+     */
+    public void setStartUserStoreEnabled(boolean enable)
+    {
+        this.startUserStore = enable;
+    }
+
     /**
      * Configure the {@link UserStore} implementation to use.
      * If none, for backward compat if none the {@link PropertyUserStore} will be used
      * @param userStore the {@link UserStore} implementation to use
      */
     public void setUserStore(UserStore userStore)
+    {
+        this.setUserStore(userStore, false);
+    }
+
+    /**
+     * Configure the {@link UserStore} implementation to use.
+     * If none, for backward compat if none the {@link PropertyUserStore} will be used
+     * @param userStore the {@link UserStore} implementation to use
+     * @param startUserStore true if doStart/doStop should call start/stop on the user store.
+     */
+    public void setUserStore(UserStore userStore, boolean startUserStore)
```

Comment:
I don't think we need this setter that takes 2 args: setters should generally only take 1 arg, and you've already got the setStartUserStoreEnabled(boolean) method.